test-data/unit/check-based-infer-function-types.test

```diff
@@ -280,6 +280,7 @@ def foo(a=1):
 
 foo()  # E: Call to untyped function "foo" in typed context  [no-untyped-call]
 reveal_type(foo)  # E: Expression type contains "Any" (has type "(int) -> Untyped")  [no-any-expr] \
+                  # E: Expression type contains "Any" (has type "(int) -> Untyped")  [no-any-expr] \
```

Comment:
this error doesnt look like it should be duplicated

Reply:
There are a lot of those defects see the upstream version of this pr: https://github.com/python/mypy/pull/12269

Reply:
maybe this change shouldn't be made until those defects are fixed

Reply:
I think it should, I think it's worth not hiding the additional errors at the cost of sometime duplicating an error on the same line.

esp for in an ide where it would be underlining only one error on a line.